buildpackrunner/runner.go

```diff
@@ -193,6 +231,37 @@ func (runner *Runner) downloadBuildpacks() error {
 	return nil
 }
 
+func (runner *Runner) cleanCacheDir() error {
+	neededCacheDirs := map[string]bool{
+		filepath.Join(runner.config.BuildArtifactsCacheDir(), "primary"): true,
+	}
+	buildpacks := runner.config.BuildpackOrder()
+	supplyBuildpacks := buildpacks[0:(len(buildpacks) - 1)]
+
+	for _, bp := range supplyBuildpacks {
+		neededCacheDirs[runner.supplyCachePath(bp)] = true
+	}
+
+	dirs, err := ioutil.ReadDir(runner.config.BuildArtifactsCacheDir())
+	if err != nil {
+		return err
+	}
+	var foundCacheDirs []string
+	for _, dirInfo := range dirs {
+		foundCacheDirs = append(foundCacheDirs, filepath.Join(runner.config.BuildArtifactsCacheDir(), dirInfo.Name()))
+	}
```

Comment:
We could just get rid of this for loop and construct the full filepath of the directory in the loop below, e.g.:
```
  for _, dirInfo := range dirs {                                                     
    dir := filepath.Join(runner.config.BuildArtifactsCacheDir(), dirInfo.Name())     
    if !neededCacheDirs[dir] {                                                       
      err = os.RemoveAll(dir)                                                        
      if err != nil {                                                                
        return err                                                                   
      }                                                                              
    }                                                                                
  }                                                                                  
```